### PR TITLE
specs changes for azure databox ga

### PR DESCRIPTION
**Pull request by sojain**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

Currently raised against preview, once signed off will move it to stable

### CODE REVIEW

File: specification/databox/resource-manager/Microsoft.DataBox/preview/2018-01-01/databox.json
Comment (anuchandy) [inline specification/databox/resource-manager/Microsoft.DataBox/preview/2018-01-01/databox.json]: You can always provide description of enum values as below:

```json
        "disabledReason": {
          "type": "string",
          "description": "Reason why the Sku is disabled.",
          "enum": [
            "None",
            "Country",
            "Region",
            "Feature",
            "OfferType",
            "Bitrate"
          ],
          "x-ms-enum": {
            "name": "SkuDisabledReason",
            "values": [
              {
                "value": "None",
                "description": "disabled due to unknown reason"
              },
              {
                "value": "Country",
                "description": "SKU is not available in the country."
              },
              {
                "value": "Feature",
                "description": "Feature is not enabled for the SKU."
              },
              ...
            ],
            "modelAsExtensible": false
          },
        },
```

**Review by anuchandy — COMMENTED** [review]
in general modelAsString:false has extensibility constrain, i.e. if service introduce a new value for the enum in the same api-version then SDKs can break. If we are not anticipating any new values in the current api-version then we are good here otherwise keep `modelAsString:false`. (for the discriminator specific enums you can keep `modelAsString:false`).

**Comment by anuchandy** [comment]
@ravbhatnagar please take a look, this is addition to preview feature. Also need your input on linked access refer [this](
https://github.com/Azure/azure-rest-api-specs/pull/3885#discussion_r217202024).

**Comment by anuchandy** [comment]
@sojain between you will need to update your readme file to have tags, like below:


# DataBox

> see https://aka.ms/autorest

This is the AutoRest configuration file for DataBox.



---
## Getting Started
To build the SDK for DataBox, simply [Install AutoRest](https://aka.ms/autorest/install) and in this folder, run:

> `autorest`

To see additional help and options, run:

> `autorest --help`
---

## Configuration

### Basic Information
These are the global settings for the DataBox API.

``` yaml
openapi-type: arm
tag: package-2018-01
```

### Tag: package-2018-01

These settings apply only when `--tag=package-2018-01` is specified on the command line.

``` yaml $(tag) == 'package-2018-01'
input-file:
- Microsoft.DataBox\preview\2018-01-01\databox.json
```

## Suppression

``` yaml
directive:
  - suppress:
    - R2016 #to suppress (PatchBodyParametersSchema/R2016/RPCViolation)
    - R2062 #to suppress (XmsResourceInPutResponse/R2062/RPCViolation)
```

---
# Code Generation


## Swagger to SDK

This section describes what SDK should be generated by the automatic system.
This is not used by Autorest itself.

``` yaml $(swagger-to-sdk)
swagger-to-sdk:
  - repo: azure-sdk-for-go
  - repo: azure-sdk-for-node
```


## C#

These settings apply only when `--csharp` is specified on the command line.
Please also specify `--csharp-sdks-folder=<path to "SDKs" directory of your azure-sdk-for-net clone>`.

``` yaml $(csharp)
csharp:
  azure-arm: true
  license-header: MICROSOFT_MIT_NO_VERSION
  namespace: Microsoft.Azure.Management.DataBox
  output-folder: $(csharp-sdks-folder)/DataBox/Management.DataBox/Generated
  clear-output-folder: true
```


## Go

These settings apply only when `--go` is specified on the command line.

``` yaml $(go)
go:
  license-header: MICROSOFT_APACHE_NO_VERSION
  namespace: databox
  clear-output-folder: true
```

### Go multi-api

``` yaml $(go) && $(multiapi)
batch:
  - tag: package-2018-01
```

### Tag: package-2018-01 and go

Please also specify `--go-sdk-folder=<path to the root directory of your azure-sdk-for-go clone>`.

``` yaml $(go)
output-folder: $(go-sdk-folder)/services/preview/databox/mgmt/2018-01-01/databox
```


## Java

These settings apply only when `--java` is specified on the command line.
Please also specify `--azure-libraries-for-java-folder=<path to the root directory of your azure-libraries-for-java clone>`.

``` yaml $(java)
java:
  azure-arm: true
  fluent: true
  namespace: com.microsoft.azure.management.databox
  license-header: MICROSOFT_MIT_NO_CODEGEN
  payload-flattening-threshold: 1
  output-folder: $(azure-libraries-for-java-folder)/databox
```

### Java multi-api

``` yaml $(java) && $(multiapi)
batch:
  - tag: package-2018-01
```

### Tag: package-2018-05 and java

These settings apply only when `--tag=package-2018-01 --java` is specified on the command line.
Please also specify `--azure-libraries-for-java-folder=<path to the root directory of your azure-sdk-for-java clone>`.

``` yaml $(tag)=='package-2018-01' && $(java) && $(multiapi)
java:
  namespace: com.microsoft.azure.management.databox.v2018_01_01
  output-folder: $(azure-libraries-for-java-folder)/databox/resource-manager/v2018_01_01
regenerate-manager: true
generate-interface: true
```

File: specification/databox/resource-manager/Microsoft.DataBox/preview/2018-01-01/databox.json
Comment (sojain) [reply specification/databox/resource-manager/Microsoft.DataBox/preview/2018-01-01/databox.json]: We have enabled linked access to storage service, and in most happy paths we only take accountId from customer thats why it is the only required field. Customers came up ith requirements that they want to copy data to National clouds, though we dont have services in that region yet and linked access from public cloud will not work on national cloud. Hence it became important to give customers the option to pass these parameters , can you suggest ways i can make it more explicit in comments

File: specification/databox/resource-manager/Microsoft.DataBox/preview/2018-01-01/databox.json
Comment (sojain) [reply specification/databox/resource-manager/Microsoft.DataBox/preview/2018-01-01/databox.json]: These properties are not used will remove the same

File: specification/databox/resource-manager/Microsoft.DataBox/preview/2018-01-01/databox.json
Comment (sojain) [reply specification/databox/resource-manager/Microsoft.DataBox/preview/2018-01-01/databox.json]: These properties are not used will remove the same

File: specification/databox/resource-manager/Microsoft.DataBox/preview/2018-01-01/databox.json
Comment (sojain) [reply specification/databox/resource-manager/Microsoft.DataBox/preview/2018-01-01/databox.json]: sure, will do

File: specification/databox/resource-manager/Microsoft.DataBox/preview/2018-01-01/databox.json
Comment (KrisBash) [inline specification/databox/resource-manager/Microsoft.DataBox/preview/2018-01-01/databox.json]: What is the migration path for deprecation of this API? It appears to be in active current use.

**Review by KrisBash — CHANGES_REQUESTED** [review]
Even though this is a preview version, these changes need to correspond to a new API version to avoid breaking customers. Some of these deleted POST apis are actively being used, and a deprecation plan is advised.

File: specification/databox/resource-manager/Microsoft.DataBox/preview/2018-01-01/databox.json
Comment (sojain) [reply specification/databox/resource-manager/Microsoft.DataBox/preview/2018-01-01/databox.json]: Till now are only active user is our own portal and we have stopped using these apis, our production systems have already stopped supporting these and see no usage/failure because of its remove. Hence as we go GA on 24th september we dont want the apis to be available for our GA sdk

File: specification/databox/resource-manager/Microsoft.DataBox/preview/2018-01-01/databox.json
Comment (sojain) [reply specification/databox/resource-manager/Microsoft.DataBox/preview/2018-01-01/databox.json]: taken appropriate fixes

File: specification/databox/resource-manager/Microsoft.DataBox/preview/2018-01-01/databox.json
Comment (sojain) [reply specification/databox/resource-manager/Microsoft.DataBox/preview/2018-01-01/databox.json]: Post more discussion with leads we have removed these changes, we will give a private sdk for handful of national cloud customers

File: specification/databox/resource-manager/Microsoft.DataBox/preview/2018-01-01/databox.json
Comment (sojain) [reply specification/databox/resource-manager/Microsoft.DataBox/preview/2018-01-01/databox.json]: nice to learn comment taken

File: specification/databox/resource-manager/Microsoft.DataBox/preview/2018-01-01/databox.json
Comment (sojain) [reply specification/databox/resource-manager/Microsoft.DataBox/preview/2018-01-01/databox.json]: Here is the plan
1) For Apis GetCopyLogsUri, DownloadShippingLabelUri, ReportIssue have zero users and the api has been disabled and non exisitent in our service for more than a month, so we are removing it from sdk also
2) ListSecrets is continued to be supported from service but is no longer support in the new generated client dll, anyone calling the api will be returned a proper response

Just FYI we have only testing (internal) sdk users, post we go GA we will have more sdk users

**Comment by sojain** [comment]
@anuchandy , i have taken most of the comments and also the readme update, however post that build is not working, can you help me in finding the root cause

File: specification/databox/resource-manager/Microsoft.DataBox/preview/2018-01-01/databox.json
Comment (KrisBash) [reply specification/databox/resource-manager/Microsoft.DataBox/preview/2018-01-01/databox.json]: @sojain There are no external users of this API? Why not take this to a new API version? These are significant changes in an API version. Are you planning to move to a stable API for your GA?

Comment (sojain) [reply specification/databox/resource-manager/Microsoft.DataBox/preview/2018-01-01/databox.json]: @KrisBash , yes this pr is to show the diffs, as soon as I have a sign off ill update this spec to stable and only post its completion we will generate a new dll for customer

**Comment by anuchandy** [comment]
@sojain I've created readme file for databox here https://raw.githubusercontent.com/anuchandy/sample-readme/master/readme.md which includes settings for all languages. Can you try this?

**Comment by anuchandy** [comment]
@sojain you also need sdks for all languages right? the readme contains entries for C# and Go only. The [link](
https://raw.githubusercontent.com/anuchandy/sample-readme/master/readme.md) i shared contains entries for all languages.

**Comment by sojain** [comment]
@anuchandy , thanks for sharing this.

This failed with 
_Command: bundle install && rake arm:regen_all_profiles['azure_mgmt_databox']
Finished with return code 10
and output:
Don't run Bundler as root. Bundler can ask for sudo if it is needed, and
installing your bundle as root will break this application for all non-root
users on this machine.
Could not locate Gemfile_

It works if a remove bundle command, any suggestions. Also just fo rmy knowledge did you some tool to generate the readme.md

**Comment by anuchandy** [comment]
@lmazuel the readme's swagger-to-sdk section looks like below:

```json
swagger-to-sdk:
  - repo: azure-sdk-for-python
  - repo: azure-sdk-for-node
  - repo: azure-sdk-for-go
  - repo: azure-sdk-for-ruby
  - repo: azure-sdk-for-java
    after_scripts:
      - bundle install && rake arm:regen_all_profiles['azure_mgmt_databox']
```

The command `bundle install && rake arm:regen_all_profiles['azure_mgmt_databox']` to generate the ruby sdk is failing with error 

```
Command: bundle install && rake arm:regen_all_profiles['azure_mgmt_databox']
Finished with return code 10
and output:
Don't run Bundler as root. Bundler can ask for sudo if it is needed, and
installing your bundle as root will break this application for all non-root
users on this machine.
Could not locate Gemfile
```

Talked to @sarangan12 about this, he mentioned that this command shouldn't be run as a root user. Can you take a look and help here?

**Comment by anuchandy** [comment]
@sojain i manually author the readme, it's not generated.

**Comment by sarangan12** [comment]
@anuchandy The error is expected. It is not because of the root user. The azure_mgmt_databox is a new gem which does not have a skeleton structure. So, it is failing. You can ignore this error

File: specification/databox/resource-manager/Microsoft.DataBox/preview/2018-01-01/databox.json
Comment (KrisBash) [reply specification/databox/resource-manager/Microsoft.DataBox/preview/2018-01-01/databox.json]: @sojain Thanks

**Review by KrisBash — APPROVED** [review]

**Comment by sojain** [comment]
thanks @anuchandy for all the help, who can help me in closing this pull request given I have arm signoff

**Comment by anuchandy** [comment]
@sojain - is this api-version deployed to at least one production region? if yes, we're good to merge. let me know.

**Comment by sojain** [comment]
@anuchandy yes api version is deployed to all regions

**Comment by sojain** [comment]
@anuchandy, sorry to bother, but can you please help in completing this pull request, we have GA at ignite on 24th and hence the urgency

**Review by anuchandy — APPROVED** [review]

**Pull request merged**
